numpy/core/numerictypes.py

```diff
@@ -210,8 +210,20 @@ def english_capitalize(s):
 
 
 sctypeDict = {}      # Contains all leaf-node scalar types with aliases
-sctypeNA = {}        # Contails all leaf-node types -> numarray type equivalences
-allTypes = {}      # Collect the types we will add to the module here
+class TypeNADict(dict):
+    def __getitem__(self, key):
+        # 2018-June-24, 1.16
+        warnings.warn('sctypeNA and typeNA will be removed in v1.17 '
+                      'of numpy', DeprecationWarning, stacklevel=2)
+        return dict.__getitem__(self, key)
+    def get(self, key, default=None):
+        # 2018-June-24, 1.16
```

Comment:
Elsewhere we use 2018-06-24, which is ISO-8601

Reply:
fixed